atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/CellTest.java

```diff
@@ -91,27 +82,18 @@ public void testHashCode() {
 
     @Test
     public void testSizeInBytes() {
-        for (Pair<Integer, Integer> sizes : allPairs(THREE_CELL_NAME_SIZES)) {
-            for (Pair<Byte, Byte> bytes : allPairs(TWO_BYTES)) {
-                assertThat(Cell.create(
-                                        spawnBytes(sizes.getLhSide(), bytes.getLhSide()),
-                                        spawnBytes(sizes.getRhSide(), bytes.getRhSide()))
-                                .sizeInBytes())
-                        .isEqualTo(sizes.getLhSide() + sizes.getRhSide());
-            }
-        }
+        assertThat(createCellWithByteSize(2).sizeInBytes()).isEqualTo(2);
+        assertThat(createCellWithByteSize(63).sizeInBytes()).isEqualTo(63);
+        assertThat(createCellWithByteSize(256).sizeInBytes()).isEqualTo(256);
     }
 
-    private static <T> Set<Pair<T, T>> allPairs(Set<T> set) {
-        return Sets.cartesianProduct(set, set).stream()
-                .map(list -> Pair.create(list.get(0), list.get(1)))
-                .collect(Collectors.toSet());
+    private static Cell createCellWithByteSize(int size) {
+        Preconditions.checkArgument(size > 1, "Size should be at least 2");
+        return Cell.create(spawnBytes(size / 2), spawnBytes(size - (size / 2)));
```

Comment:
Maybe I'm being a bit thick, but shouldn't `size / 2` == `size - (size / 2)` the vast majority of the time?

Reply:
They are equal when n is even

Reply:
why do you want them to not be equal when `n` is odd?

Reply:
I think @ergo14 wants the cell to have exactly the right size (e.g., otherwise `createCellWithByteSize(7)` returns a cell that has row and column of size `3`, i.e. an overall size of `6`) - it'll be one byte smaller if we just divide